### PR TITLE
chore: Bump ginkgo version

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind support

**What this PR does / why we need it**:
This PR bumps ginkgo to v2.8.4. Also removed the dependency for ginkgo from /hack/tools. Refer [here](https://github.com/onsi/ginkgo/releases/tag/v2.8.2) for more details.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Ankitasw** [comment]
@richardcase @Skarlso can someone help fix this error :(

**Comment by Skarlso** [comment]
I'll take a look. 
/assign

File: Makefile [hunk 1]
```diff
@@ -398,28 +397,28 @@ test-verbose: setup-envtest ## Run tests with verbose settings.
...
-test-e2e: $(GINKGO) $(KIND) $(SSM_PLUGIN) $(KUSTOMIZE) generate-test-flavors e2e-image ## Run e2e tests
-	time $(GINKGO) -tags=e2e $(GINKGO_ARGS) -p ./test/e2e/suites/unmanaged/... -- -config-path="$(E2E_CONF_PATH)" $(E2E_ARGS)
+test-e2e: $(KIND) $(SSM_PLUGIN) $(KUSTOMIZE) generate-test-flavors e2e-image ## Run e2e tests
+	time go run github.com/onsi/ginkgo/v2/ginkgo -tags=e2e $(GINKGO_ARGS) -p ./test/e2e/suites/unmanaged/... -- -config-path="$(E2E_CONF_PATH)" $(E2E_ARGS)
 
 .PHONY: test-e2e-eks ## Run EKS e2e tests using clusterctl
-test-e2e-eks: generate-test-flavors  $(GINKGO) $(KIND) $(SSM_PLUGIN) $(KUSTOMIZE) e2e-image ## Run eks e2e tests
-	time $(GINKGO) -tags=e2e $(GINKGO_ARGS) ./test/e2e/suites/managed/... -- -config-path="$(E2E_EKS_CONF_PATH)" --source-template="$(EKS_SOURCE_TEMPLATE)" $(E2E_ARGS) $(EKS_E2E_ARGS)
+test-e2e-eks: generate-test-flavors $(KIND) $(SSM_PLUGIN) $(KUSTOMIZE) e2e-image ## Run eks e2e tests
+	time go run github.com/onsi/ginkgo/v2/ginkgo-tags=e2e $(GINKGO_ARGS) ./test/e2e/suites/managed/... -- -config-path="$(E2E_EKS_CONF_PATH)" --source-template="$(EKS_SOURCE_TEMPLATE)" $(E2E_ARGS) $(EKS_E2E_ARGS)
```
Comment (Skarlso) [inline Makefile:405]:
```suggestion
	time go run github.com/onsi/ginkgo/v2/ginkgo -tags=e2e $(GINKGO_ARGS) ./test/e2e/suites/managed/... -- -config-path="$(E2E_EKS_CONF_PATH)" --source-template="$(EKS_SOURCE_TEMPLATE)" $(E2E_ARGS) $(EKS_E2E_ARGS)
```

**Comment by Skarlso** [comment]
Removing the `tools` build tag from golangci-lint will fix the lint issue. Testing doesn't use the `tools` tag as far as I can see.

**Comment by Skarlso** [comment]
We can bump to the next version according to this comment: https://github.com/onsi/ginkgo/issues/1149#issuecomment-1445129397

**Comment by Ankitasw** [comment]
@Skarlso I will try it out on Monday. Thanks @Skarlso

**Comment by Ankitasw** [comment]
PR is ready for review.

**Comment by Ankitasw** [comment]
@Skarlso can I get 👀 on this 🙂 ?

**Pull request merged**